operator/pkg/manifest/installer.go

```diff
@@ -267,19 +268,48 @@ func parseKubectlVersion(kubectlStdout string) (*goversion.Version, *goversion.V
 }
 
 // ApplyAll applies all given manifests using kubectl client.
-func ApplyAll(manifests name.ManifestMap, version pkgversion.Version, iop *v1alpha1.IstioOperatorSpec, opts *kubectlcmd.Options) (CompositeOutput, error) {
+func ApplyAll(manifests name.ManifestMap, version pkgversion.Version, iop *v1alpha1.IstioOperatorSpec,
+	opts *kubectlcmd.Options, certDir, clusterID string) (CompositeOutput, error) {
 	scope.Infof("Preparing manifests for these components:")
 	for c := range manifests {
 		scope.Infof("- %s", c)
 	}
 	scope.Infof("Component dependencies tree: \n%s", installTreeString())
-	if _, err := InitK8SRestClient(opts.Kubeconfig, opts.Context); err != nil {
+	restConfig, err := InitK8SRestClient(opts.Kubeconfig, opts.Context)
+	if err != nil {
+		return nil, err
+	}
+	client, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
 		return nil, err
 	}
 	// Set up the namespace for installation
 	if err := createNamespace(iop.Namespace); err != nil {
 		return nil, err
 	}
+
+	if clusterID == "" {
```

Comment:
A few comments:

- clusterID is only needed if certDir is set. The clusterID selection can be moved under the `certDir != ""` conditional.
- `--cert-dir` is new behavior so it may be okay to return an error if we can't pick a unique default clusterID.
- This logic should all probably be extracted out to a helper function (for example, `plugExternalCerts()`).

Reply:
done